### PR TITLE
Fix store opening from the search box

### PR DESCRIPTION
**Pull request by huwshimi**
Fixes: #3313.

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- visit /u/hatch (or another profile with charms)
- click on a charm and then click View
- the charm details should appear (not the base store page)
- click add to canvas
- the charm should be added to the canvas and the store shouldn't open

**Comment by huwshimi** [comment]
[Fix store state when deploying/viewing entities from a profile.](https://trello.com/c/yTaUhsVS/21-fix-store-state-when-deploying-viewing-entities-from-a-profile)

**Comment by jujugui** [comment]
Build finished.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8260/

File: jujugui/static/gui/src/app/components/user-profile/entity/entity.js [hunk 1]
```diff
@@ -98,8 +95,6 @@ class UserProfileEntity extends React.Component {
 
   /**
     Generate a list of commits.
-
-    @method _generateTags
```
Comment (frankban) [inline jujugui/static/gui/src/app/components/user-profile/entity/entity.js]: Thanks for this clean up.

File: jujugui/static/gui/src/app/components/header-search/header-search.js [hunk 2]
```diff
@@ -100,32 +100,34 @@ class HeaderSearch extends React.Component {
...
+      if (!appState.current.store) {
+        newState.store = '';
+      }
       if (!appState.current.model) {
         newState.root = 'new';
-      };
+      }
       appState.changeState(newState);
     }
+    // Opening the search needs to be after the above condition as it modifies
```
Comment (frankban) [inline jujugui/static/gui/src/app/components/header-search/header-search.js]: Cool.

**Comment by frankban** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**